pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -37,13 +40,33 @@ def _target_satellite_host(request, satellite_factory):
         yield
 
 
+@lru_cache
+def capsule_cdn_register(hostname=None):
+    cap = Capsule(hostname=hostname)
+    # Enabling CDN repos on Capsule Host
+    cap.register_to_cdn()
+    # Enabling OHSnap repos on Capsule host
+    for repo in getattr(constants, f"OHSNAP_RHEL{cap.os_version.major}_REPOS"):
+        result = cap.enable_repo(repo, force=True)
+        if result.status:
+            raise CapsuleHostError(f'Repo enable at capsule host failed\n{result.stdout}')
+
+
 @contextmanager
 def _target_capsule_host(request, capsule_factory):
-    if 'sanity' not in request.config.option.markexpr:
+    if 'sanity' not in request.config.option.markexpr and not request.config.option.n_minus:
         new_cap = capsule_factory()
         yield new_cap
         new_cap.teardown()
         Broker(hosts=[new_cap]).checkin()
+    elif request.config.option.n_minus:
+        if not settings.capsule.hostname:
+            hosts = Capsule.get_hosts_from_inventory(filter="'cap' in @inv.name")
+            settings.capsule.hostname = hosts[0].hostname
```

Comment:
We are in CI are spinning only one pair of Satellite and Capsule, hence there will always be only one satellite and capsule. 

For local setting one will always pass capsule hostname in capsule.yaml conf !

Reply:
Got it. If I understand it correctly, this is something that could be considered as an improvement for the future work on capsule n minus.

Reply:
Yes, To improvise the time or concurrency we could probably go for team specific jobs in CI OR multiple capsules where test has to be mapped with the particular capsule so that the entire test would run on same capsule and not split into multiple.

BTW, that is not yet the need in near future, if its needed we will reevaluate.